### DISCUSSION TITLE
PROPOSAL: Convention for representing abstraction / generalization in paths et al

### DISCUSSION THREAD

**Discussion by ClaytonCurtis1947** [freeplane/freeplane#1234] [Ideas]
Based on comments from the community, there does not seem to be a convention for documentation styling to represent abstraction / generalization. Here I present examples of the need for one and a modest proposal.

CASE IN POINT: At 'https://docs.freeplane.org/user-documentation/templates.html' there is a space for Windows paths to system and user-defined templates that currently reads "TODO', and I'm trying to fill that gap.

The path descriptions for Linux are pretty straightforward:

= = = = =
**System-wide** templates (when Freeplane is installed via the package manager of your distribution) are located in:

`/usr/share/freeplane/resources/templates`

**User-defined** templates can be found in:

`~/.config/freeplane/VERSION/templates`
= = = = =

For Windows, example equivalents in concrete form are:
`\Program Files\Freeplane\resources\templates`
and
`\Users\curti\AppData\Roaming\Freeplane\1.11.x\templates` <<<<< I'm the 'curti'

CHALLENGE: How does one represent generalization or abstraction of a path element. For example, is '/VERSION/' in the Linux case actual text that would be used or an abstraction of the concept of version?
 
For Windows, there is no abstraction for version. There is also no abstraction for user other than plugging in an environment variable such as 'USERNAME', which would be represented in a path as '\Users%USERNAME%'. So the best approximation is

`\Users%USERNAME%\AppData\Roaming\Freeplane\1.11.x\templates` -- which still uses an explicit specification of version and leaves open the question of representing an abstraction.

PROPOSAL: All of the usual bracketing characters run the risk of being used explicitly in one context or another. Therefore I propose taking advantage of actual text styling to indicate an abstraction -- italicized and bold text.  [I'd rather have used underlined, but markdown doesn't support that due to potential confusion with hyperlinks.]

EXAMPLE: The above Windows path would be represented by

`\\users\\***user***\AppData\Roaming\Freeplane\\***version***\templates`

NOTE: I'm not one who believes there have to be rigid rules for everything. I'm just proposing a =convention= that I can use to address a problem and others can adopt if they wish.  I absolutely welcome alternative solutions!!
		
Thanks!

**Comment by macmarrum** [comment]
Hi @ClaytonCurtis1947 

That's a good point. It is a challenge to make it clear that a part of a path is variable.

So far paths have been presented as _code_, i.e.
```markdown
`~/.config/freeplane/VERSION/templates`
```
Using this approach, no (additional) formatting can be applied.

As far as I understand, you are proposing to no longer use _code_ for paths and introduce bold+italics to indicate special meaning (a variable), e.g.

~/.config/freeplane/**_VERSION_**/templates

Looking at both lines, I must say that I find the _code_ styling (first one) better.

I also recall a convention where variables in paths are marked as `<VARIABLE>`, i.e.
`~/.config/freeplane/<VERSION>/templates`
`C:\Users\<USERNAME>\AppData\Roaming\Freeplane\<VERSION>\templates`

I like this approach best, because it indicates special meaning and still allows to use _code_ for paths.

How would you feel about using this convention?
